File: apps/csharp/design-pattern/tests/DesignPattern.Tests/Patterns/AdapterTest.cs
using DesignPattern.Patterns.Adapter;

namespace DesignPattern.Tests.Patterns;

public class AdapterTest
{
    [Fact]
    public void Adapter_ConvertsTextProperty()
    {
        var british = new BritishTextObject { String = "Hello" };
        ITextObject adapted = new BritishTextObjectAdapter(british);

        Assert.Equal("Hello", adapted.Text);
    }

    [Fact]
    public void Adapter_ConvertsMmToInches()
    {
        var british = new BritishTextObject { SizeMm = 25.4 };
        ITextObject adapted = new BritishTextObjectAdapter(british);

        Assert.Equal(1, adapted.SizeInches);
    }

    [Fact]
    public void Adapter_ConvertsInchesToMm()
    {
        var british = new BritishTextObject();
        var adapted = new BritishTextObjectAdapter(british);

        adapted.SizeInches = 2;

        Assert.Equal(50.8, british.SizeMm, 1);
    }

    [Fact]
    public void Adapter_ConvertsColorProperty()
    {
        var british = new BritishTextObject { Colour = "red" };
        ITextObject adapted = new BritishTextObjectAdapter(british);

        Assert.Equal("red", adapted.Color);
    }

    [Fact]
    public void Renderer_CanRenderAdaptedObject()
    {
        var british = new BritishTextObject
        {
            String = "Adapted",
            SizeMm = 50.8,
            Colour = "blue"
        };
        var adapted = new BritishTextObjectAdapter(british);
        var renderer = new Renderer();

        var result = renderer.Render(adapted);

        Assert.Contains("Adapted", result);
        Assert.Contains("2in", result);
        Assert.Contains("blue", result);
    }

    [Fact]
    public void Adapter_SetsTextOnAdaptee()
    {
        var british = new BritishTextObject();
        var adapted = new BritishTextObjectAdapter(british);

        adapted.Text = "Updated";

        Assert.Equal("Updated", british.String);
    }
}

File: apps/php/design-pattern/tests/AdapterTest.php
<?php

declare(strict_types=1);

namespace DesignPattern\Tests;

use DesignPattern\Adapter\BritishTextObject;
use DesignPattern\Adapter\BritishTextObjectAdapter;
use DesignPattern\Adapter\Renderer;
use PHPUnit\Framework\TestCase;

class AdapterTest extends TestCase
{
    public function testAdapterGetText(): void
    {
        $british = new BritishTextObject('Hello', 25.4, 'red');
        $adapter = new BritishTextObjectAdapter($british);

        $this->assertSame('Hello', $adapter->getText());
    }

    public function testAdapterConvertsMmToInches(): void
    {
        $british = new BritishTextObject('Hello', 25.4, 'red');
        $adapter = new BritishTextObjectAdapter($british);

        $this->assertEqualsWithDelta(1.0, $adapter->getSizeInches(), 0.001);
    }

    public function testAdapterGetColor(): void
    {
        $british = new BritishTextObject('Hello', 25.4, 'blue');
        $adapter = new BritishTextObjectAdapter($british);

        $this->assertSame('blue', $adapter->getColor());
    }

    public function testRendererWithAdapter(): void
    {
        $british = new BritishTextObject('Test', 50.8, 'green');
        $adapter = new BritishTextObjectAdapter($british);
        $renderer = new Renderer();

        $result = $renderer->render($adapter);

        $this->assertStringContainsString('Test', $result);
        $this->assertStringContainsString('2.0 inches', $result);
        $this->assertStringContainsString('green', $result);
    }

    public function testBritishTextObjectDirectAccess(): void
    {
        $british = new BritishTextObject('Direct', 10.0, 'yellow');

        $this->assertSame('Direct', $british->getString());
        $this->assertSame(10.0, $british->getSizeMm());
        $this->assertSame('yellow', $british->getColour());
    }
}
